### PR TITLE
V5 migration e2e

### PR DESCRIPTION
**Pull request by leungkinghin**
End-to-end tests for 5 payment methods have been migrated : 
- credit card
- credit card redirect
- credit card 3DS native
- Klarna
- Affirm

Description of E2E test flow
1. Create the session with appropriate request (e.g. klarna payment requires line items, affirm payment only supports US location)
2. Obtain session ID and session data from response. Use them to initialize the AdyenCheckout web component
https://docs.adyen.com/online-payments/web-components#set-up
3. For credit card, we can obtain the result code from onPaymentCompleted function right away. Put the result code in text box in HTML page. It is retrieved by test case and do the assertion.
4. For other payment methods, authentication page is displayed. After submitting credentials, external payment gateway would redirect the page /return-url
5. In return-url page, we obtain the session ID and redirect result. We would pass this two information into `redirect-payment-form.html` and then submit the finalized request to Adyen
https://docs.adyen.com/online-payments/web-drop-in#handleredirectresult
7. Get the final result code from the `onPaymentCompleted` function in `redirect-payment-form.html` and then do the assertion to verify the payment is completed successfully.

p.s. Manual Capture for Klarna payment can only be done after we resolved webhook setup in CI for e2e test.

### CODE REVIEW

File: extension/test/e2e/affirm.spec.js [hunk 1]
```diff
@@ -65,118 +68,94 @@ describe.skip('::affirmPayment::', () => {
...
+          ctpClient,
+          adyenMerchantAccount,
+          ctpProjectKey,
+          createSessionRequest,
+          'USD'
+        )
         logger.debug(
-          'affirm::paymentAfterMakePayment:',
-          JSON.stringify(paymentAfterMakePayment)
+          'credit-card::paymentAfterCreateSession:',
```
Comment (praveenkumarct) [inline extension/test/e2e/affirm.spec.js:91]: should it be affirm::paymentAfterCreateSession?

File: extension/test/e2e/pageObjects/AffirmInitSessionFormPage.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  async initPaymentSession({ clientKey, paymentAfterCreateSession }) {
+    await super.initPaymentSession(clientKey, paymentAfterCreateSession)
+    await this.pasteValuesInAffirmWebComponent()
+    await this.confirmAffirmWebComponent()
+  }
+
+  async pasteValuesInAffirmWebComponent() {
+    await this.page.waitForSelector('.adyen-checkout__button--pay') // wait for rendering of web component
+
+    const createSessionInputFormElement = await this.page.$(
```
Comment (praveenkumarct) [inline extension/test/e2e/pageObjects/AffirmInitSessionFormPage.js:13]: I think we could rename **createSession** here, the naming can be something else like: **adyenCheckoutInputFormElement**?

File: extension/test/e2e/affirm.spec.js (hunk 1 above)
Comment (leungkinghin) [reply extension/test/e2e/affirm.spec.js]: Good catch
https://github.com/commercetools/commercetools-adyen-integration/pull/1064/commits/946c4d14c46c956e4250035afa1ff1c51d36c544

File: extension/test/e2e/pageObjects/AffirmInitSessionFormPage.js (hunk 2 above)
Comment (leungkinghin) [reply extension/test/e2e/pageObjects/AffirmInitSessionFormPage.js]: https://github.com/commercetools/commercetools-adyen-integration/pull/1064/commits/a77571e0a5e21dfb1fd8da8470085809f70df006

**Review by praveenkumarct — APPROVED** [review]

**Pull request merged**